internal/providers/pke/pkeworkflow/update_node_group_activity.go

```diff
@@ -0,0 +1,277 @@
+// Copyright © 2020 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pkeworkflow
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"emperror.dev/errors"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	"go.uber.org/cadence"
+	"go.uber.org/cadence/activity"
+
+	cloudformation2 "github.com/banzaicloud/pipeline/internal/cloudformation"
+	pkgCluster "github.com/banzaicloud/pipeline/internal/cluster"
+	"github.com/banzaicloud/pipeline/pkg/brn"
+	pkgCloudFormation "github.com/banzaicloud/pipeline/pkg/providers/amazon/cloudformation"
+	sdkAmazon "github.com/banzaicloud/pipeline/pkg/sdk/providers/amazon"
+	sdkCloudFormation "github.com/banzaicloud/pipeline/pkg/sdk/providers/amazon/cloudformation"
+)
+
+const awsNoUpdatesError = "No updates are to be performed."
+
+const UpdateNodeGroupActivityName = "pkeaws-update-node-group"
```

Comment:
**Note**: in this package the consistent activity naming is `pke-<activity>`. Not sure what would be best here, I can accept being more strict in this one.

Reply:
This is an interesting case, I would use the newly established `pke-aws-` naming here because that should have been used here as well in the first case. The general rule of thumb was that everything which is simple "pke" is the AWS implementation, but that was a bad idea in the original implementation.

I think we should use the pke-aws- naming and reregister the activities also with that name later on in this package (but keeping the old name as well).

Reply:
I agree with your reasoning.